gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/DeadlineEnforcementDagProc.java

```diff
@@ -68,7 +68,7 @@ protected boolean isDagStillPresent(Optional<Dag<JobExecutionPlan>> dag, DagMana
 
     if (!dagManagementStateStore.existsJobDagAction(dagAction.getFlowGroup(), dagAction.getFlowName(),
         dagAction.getFlowExecutionId(), dagAction.getJobName(), dagAction.getDagActionType())) {
-      log.warn("Dag action {} is cleaned up from DMSS. No further action is required.", dagAction);
+      log.info("Dag action {} is cleaned up from DMSS. No further action is required.", dagAction);
```

Comment:
what's your take [here](https://github.com/apache/gobblin/pull/4012#discussion_r1701588762)?

to repeat:
> don't we already hold a lease by the time act is called? if so, it would be exceedingly rare (and really no different than any other DagProc::act derived class impl). given the rarity, I'd suggest to remove.
>
> but if we really believe we need this, let's observe the commonality by placing this impl within the DagProc base class

Reply:
this is deadline dag proc. the reevaluate dag proc could also clean the same dag. so basically it is about the contention between different type of dag procs, so lease will not help here

Reply:
i would say lets keep it here, because this is the only case where one dag action can be cleaned up by other type of dag proc.
otherwise maintainers will keep wondering if there is some general conditions, irrespective of the type of dag actions, that can cause this